python/pyspark/tests/test_rdd.py

```diff
@@ -814,6 +814,63 @@ def assert_request_contents(exec_reqs, task_reqs):
         rddWithoutRp = self.sc.parallelize(range(10))
         self.assertEqual(rddWithoutRp.getResourceProfile(), None)
 
+    def test_collect_with_job_group(self):
```

Comment:
Shall we mimic the test case at https://github.com/apache/spark/pull/24898/files#diff-896aaa74767987ed14f1ef9fcb324cefR84?